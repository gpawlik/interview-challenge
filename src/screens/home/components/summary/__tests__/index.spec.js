import * as React from "react";
import renderer from "react-test-renderer";

import { Summary } from "../";

jest.mock("react-redux", () => ({
  useSelector: (item) => item(),
}));

jest.mock("../../../../../components/badge", () => ({
  Badge: "Badge",
}));

jest.mock("../../../../../state/menu/selectors", () => ({
  getSelectedItemsCount: () => 5,
  getSummaryArray: () => {
    return [
      { label: "aa", count: 2 },
      { label: "bb", count: 3 },
      { label: "cc", count: 1 },
    ];
  },
}));

jest.mock("../styles", () => ({
  SummaryLeft: "SummaryLeft",
  SummaryRight: "SummaryRight",
}));

const initialProps = {};

const mockComponent = (props) => {
  const finalProps = {
    ...initialProps,
    ...props,
  };

  return renderer.create(<Summary {...finalProps} />);
};

describe("Summary", () => {
  it("renders correctly default version", () => {
    const tree = mockComponent().toJSON();

    expect(tree).toMatchSnapshot();
  });
});
