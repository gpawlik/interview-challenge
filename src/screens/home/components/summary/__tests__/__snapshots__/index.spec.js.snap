// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary renders correctly default version 1`] = `
Array [
  <SummaryLeft
    className="col-6"
>
    <span>
        5 items
    </span>
</SummaryLeft>,
  <SummaryRight
    className="col-6"
>
    <span>
        2
        x 
        <Badge
            text="aa"
        />
    </span>
    <span>
        3
        x 
        <Badge
            text="bb"
        />
    </span>
    <span>
        1
        x 
        <Badge
            text="cc"
        />
    </span>
</SummaryRight>,
]
`;
